MGMT-12691 add support for KERNEL_VERSION etc replacement for filesToSign and unsignedImage fields (#336)

chr15p opened the pull request:
Fixes #336

Upstream-Commit: 546c83387da884997298e4b34086acd9bcbfba02
Upstream-Commit: 00d1995fd5c09fa7efd6a2630604dd074515ee58

> [MGMT-12691](https://issues.redhat.com//browse/MGMT-12691) add support for KERNEL_VERSION etc replacement for filesToSign and unsignedImage fields (#187)
> 
> filesToSign (the filepaths of the kmods to sign) and unsignedImage (the name of the image that contains the
> kmods if its not built in cluster) both normally would contain the kernel version, this commit allows the
> embedding of "${KERNEL_VERSION}" and/or ${KERNEL_FULL_VERSION} in them and have it replaced with the
> actual kernel version
> 
> The code for doing this search/replace has been added to internal/utils/replaceStrings.go to make a generally
> useful library for other places that may need similar functionality

qbarrand commented:
@chr15p how about including https://github.com/kubernetes-sigs/kernel-module-management/commit/00d1995fd5c09fa7efd6a2630604dd074515ee58 as well?
If you agree, please include the following lines in the commit message:
```
Upstream-Commit: 546c83387da884997298e4b34086acd9bcbfba02
Upstream-Commit: 00d1995fd5c09fa7efd6a2630604dd074515ee58
```

qbarrand commented:
> getting the code out of sync

Hopefully the sync tool should help there. The two `Upstream-Commit` markups help it realize that this midstream commit covers the two upstream ones. Just wanted to avoid the hassle of another PR for a very related commit, but definitely we can go with whatever we're more comfortable with.

chr15p commented:
> Just wanted to avoid the hassle of another PR for a very related commit,

Yes, these should really have been one commit in upstream so it makes sense to merge them here (especially if it avoids having to fix another cherry-pick issue), which is why I did it.  My caution is that every time we do something different in this repo to what we do upstream it risks increasing the amount of management this repo needs going forward.

With servers we'd talk about "pets verses cattle", (pets are individuals and need individual care and attention, cattle you manage as a herd so each individual takes a lot less effort) not sure what the equivalent is for repos.

Anyway, its done :) 
(hmm I changed the commit message, but probably should have changed the PR description as well... my bad, fixed)

qbarrand commented:
/lgtm

chr15p commented:
/retest
